lib/ruby_lsp/ruby_lsp_rails/definition.rb

```diff
@@ -51,53 +51,92 @@ def on_string_node_enter(node)
-      def handle_possible_dsl(node)
-        node = @node_context.call_node
-        return unless node
+      #: (Prism::CallNode node) -> void
+      def on_call_node_enter(node)
         return unless self_receiver?(node)
 
         message = node.message
 
         return unless message
 
+        if message.end_with?("_path") || message.end_with?("_url")
+          handle_route(node)
+        end
+      end
+
+      private
+
+      #: ((Prism::SymbolNode | Prism::StringNode) node) -> void
+      def handle_possible_dsl(node)
+        call_node = @node_context.call_node
+        return unless call_node
+        return unless self_receiver?(call_node)
+
+        message = call_node.message
+
+        return unless message
+
         if Support::Associations::ALL.include?(message)
-          handle_association(node)
+          handle_association(call_node)
         elsif Support::Callbacks::ALL.include?(message)
           handle_callback(node)
+          handle_if_unless_conditional(node)
+        elsif Support::Validations::ALL.include?(message)
+          handle_validation(node)
+          handle_if_unless_conditional(node)
         end
       end
 
-      #: (Prism::CallNode node) -> void
-      def on_call_node_enter(node)
-        return unless self_receiver?(node)
+      #: ((Prism::SymbolNode | Prism::StringNode) node) -> void
+      def handle_callback(node)
+        call_node = @node_context.call_node
+        return unless call_node
 
-        message = node.message
+        arguments = call_node.arguments&.arguments
+        return unless arguments
 
-        return unless message
+        focus_argument = arguments.find do |argument|
+          true if argument == node
+        end
 
-        if message.end_with?("_path") || message.end_with?("_url")
-          handle_route(node)
+        name = case focus_argument
+        when Prism::SymbolNode
+          focus_argument.value
+        when Prism::StringNode
+          focus_argument.content
         end
+
+        return unless name
+
+        collect_definitions(name)
       end
 
-      private
+      #: ((Prism::SymbolNode | Prism::StringNode) node) -> void
+      def handle_validation(node)
+        call_node = @node_context.call_node
+        return unless call_node
 
-      #: (Prism::CallNode node) -> void
-      def handle_callback(node)
-        arguments = node.arguments&.arguments
-        return unless arguments&.any?
+        message = call_node.message
+        return unless message
+
+        arguments = call_node.arguments&.arguments
+        return unless arguments
+
+        focus_argument = arguments.find do |argument|
+          argument == node
+        end
+        return unless focus_argument
 
-        arguments.each do |argument|
-          name = case argument
-          when Prism::SymbolNode
-            argument.value
-          when Prism::StringNode
-            argument.content
-          end
+        return unless node.is_a?(Prism::SymbolNode)
```

Comment:
Do validations not accept a string argument?

Reply:
I implemented only for symbols because, even though strings maybe work, through all the documentation and rails API docs all the examples are given with symbols, so maybe some parts work with strings _and_ symbols. But i prefer to enforce symbol usage, this is closer to what the Rails guides and API docs shows. With that we are sure that the go to definition are correct because the symbols are widely accepted and documented, but the string don't.